tests/zenko_tests/node_tests/backbeat/tests/lifecycle/transition.js

```diff
@@ -11,61 +13,122 @@ const keyPrefix = `${uuid()}-key-prefix`;
 
 const cloudServer = new LifecycleUtility(scalityS3Client)
-    .setBucket(srcBucket)
-    .setKeyPrefix(keyPrefix)
-    .setKey(key)
-    .setDestinationLocation(process.env.AWS_BACKEND_DESTINATION_LOCATION);
+      .setBucket(srcBucket)
+      .setKeyPrefix(keyPrefix)
+      .setKey(key);
 
-const s3 = new LifecycleUtility(awsS3Client)
-    .setBucket(process.env.AWS_CRR_BUCKET_NAME)
-    .setKeyPrefix(keyPrefix)
-    .setKey(`${srcBucket}/${key}`);
+const cloud = new LifecycleUtility(awsS3Client, sharedBlobSvc, gcpStorage)
+      .setKeyPrefix(keyPrefix)
+      .setKey(`${srcBucket}/${key}`);
 
 function compareTransitionedData(cb) {
     return series([
         next => cloudServer.getObject(next),
         next => cloudServer.putBucketLifecycleConfiguration(new Date(), next),
         next => cloudServer.waitUntilTransitioned(next),
-        next => s3.getObject(next),
+        next => cloud.getObjectDataFromLocation(next),
+        next => cloudServer.getObject(next),
     ], (err, data) => {
         if (err) {
             return cb(err);
         }
-        cloudServer._compareObjectBody(data[0].Body, data[3].Body);
+        // check data stored on the target cloud
+        cloudServer._compareObjectBody(data[0].Body, data[3]);
+        // check that object is still readable from Zenko
+        cloudServer._compareObjectBody(data[0].Body, data[4].Body);
         return cb();
     });
 };
 
-describe('transition to AWS backend', function() {
-    this.timeout(LIFECYCLE_TIMEOUT);
+const locationParams = {
+    local: {
+        supportsVersioning: true,
+    },
+    AWS: {
+        name: process.env.AWS_BACKEND_DESTINATION_LOCATION,
+        bucket: process.env.AWS_CRR_BUCKET_NAME,
+        supportsVersioning: true,
+    },
+    GCP: {
+        name: process.env.GCP_BACKEND_DESTINATION_LOCATION,
+        bucket: process.env.GCP_CRR_BUCKET_NAME,
+        supportsVersioning: false,
+    },
+    Azure: {
+        name: process.env.AZURE_BACKEND_DESTINATION_LOCATION,
+        bucket: process.env.AZURE_CRR_BUCKET_NAME,
+        supportsVersioning: false,
+    },
+};
 
-    afterEach(done => series([
-        next => cloudServer.deleteVersionedBucket(srcBucket, next),
-        next => s3.clearBucket(next),
-    ], done));
+[{
+    from: 'local',
+    to: 'AWS',
+}, {
+    from: 'local',
+    to: 'GCP',
+}, {
+    from: 'local',
+    to: 'Azure',
+}, {
+    from: 'AWS',
+    to: 'GCP',
+}, {
+    from: 'GCP',
+    to: 'Azure',
+}, {
+    from: 'Azure',
+    to: 'AWS',
+}].forEach(test => {
+    describe(`transition from ${test.from} to ${test.to}`, function() {
+        this.timeout(LIFECYCLE_TIMEOUT);
+        const fromLoc = locationParams[test.from];
+        const toLoc = locationParams[test.to];
 
-    describe('without versioning', () => {
-        beforeEach(done => cloudServer.createBucket(srcBucket, done));
+        before(() => {
+            if (test.from !== 'local') {
+                cloudServer.setSourceLocation(fromLoc.name);
```

Comment:
I think we should always set this (i.e. also in the 'Zenko' case), and then let the methods handle any logic related to the location type. That way we can benefit more from method chaining.